### PR TITLE
MGDOBR-1003: Cover the status pop up dialog

### PR DESCRIPTION
**Pull request by sjamboro**
https://issues.redhat.com/browse/MGDOBR-1003

### CODE REVIEW

**Review by jomarko — COMMENTED** [review]
Hi @sjamboro I see you have refactored `InstancesTest`. nice work. I would like to ask, as part of your PR, which tests were just moved and which are updated/totally new. I would like to review just the second group of tests.

**Comment by sjamboro** [comment]
@jomarko There should not be a new test. The popover replace accepted/preparing status of instance. So whenever a test assert "Accepted" or "Preparing" states I invoke the popover and check the actual state (not just "Creating...").

You can see it in this commit: https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/pull/188/commits/26b15e1f58460523f2835ead9715fbf1b4b13110


The second commit (https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/pull/188/commits/7f5c9d44f903cfc2aef04ff00a4837c28a09dcc6) is only about refactoring because the file was a long and the orientation was very dificult.

**Comment by sjamboro** [comment]
@jomarko for example there are assertation of new popover:
https://github.com/sjamboro/sandbox-ui/blob/MGDOBR-1003/cypress/integration/instance/InstanceCreation.ts#L47-L76
https://github.com/sjamboro/sandbox-ui/blob/MGDOBR-1003/cypress/integration/instance/InstancePage.ts#L97-L124
https://github.com/sjamboro/sandbox-ui/blob/MGDOBR-1003/cypress/integration/instance/InstanceStatuses.ts#L55-L79
https://github.com/sjamboro/sandbox-ui/blob/MGDOBR-1003/cypress/integration/instance/InstanceStatuses.ts#L116-L141

**Comment by jomarko** [comment]
Thank you for explanation @sjamboro and pointers to assertions.


As you mention, your motivation for refactoring was the length of code, I have this topic for discussion.

Currently asserting `Creating`, means to use three sub assertions, to check `Pending`, `Preparing` and `Provisioning`

```
cy.ouiaId("steps", "QE/StackItem").within(() => {
              progressStepStatus(
                SEStepStatus.Info,
                "pending",
                "Creation pending"
              );
              progressStepStatus(
                SEStepStatus.Default,
                "preparing",
                "Preparing"
              );
              progressStepStatus(
                SEStepStatus.Default,
                "provisioning",
                "Provisioning"
              );
            });            
```

Question, can not we refactor `progressStepStatus` to take single parameter? 'Pending', 'Preparing' or `Provisioning`? The the assertion above would be changed to

```
            cy.ouiaId("steps", "QE/StackItem").within(() => {
              progressStepStatus(
                "Creation pending"
              );
            });
```

**Review by Srihari1192 — APPROVED** [review]
LGTM

**Comment by sjamboro** [comment]
@jomarko do you think something like this?

**Review by jomarko — APPROVED** [review]
@sjamboro yes, I like this improvement! Thank you. well done.

**Comment by sjamboro** [comment]
@jomarko @ChughShilpa the conflict should be resolved. Please check if your tests are running properly.
Instance three -> InstanceStatuses.txs
Error Handling -> InstancePage.txs

**Comment by sjamboro** [comment]
@kelvah can you review this PR. I hope that it should be the final state. By the way, is there any chance to change the duration of the state when the instance is creating (I mean mocked data)? See my TODO comment in the InstanceCreation.ts file.

File: cypress/integration/instance/InstanceCreation.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        cy.ouiaId("info-banner", "QE/StackItem")
+          .ouiaId("ready-shortly", "PF4/Text")
+          .should("be.visible");
+        cy.ouiaId("steps-count", "QE/StackItem").should(
+          "have.text",
+          "0 of 3 steps completed"
+        );
+        //TODO: There should be a check that all steps are met but I did not know how to do it for Mocked Env
+        //It seems that '1 of 3' is skipped and `2 of 3` is here only for few secs.
+        //Each step is asserted with the mocked version
```
Comment (kelvah) [inline cypress/integration/instance/InstanceCreation.ts:60]: I think we can improve the mocked API handler to also include the skipped `preparing` status. Could you please create a JIRA and link it in this comment?

**Review by kelvah — APPROVED** [review]
LGTM

File: cypress/integration/instance/InstanceCreation.ts (hunk 1 above)
Comment (sjamboro) [reply cypress/integration/instance/InstanceCreation.ts:60]: Yes ... https://issues.redhat.com/browse/MGDOBR-1118

**Pull request merged**